### ISSUE TITLE
[Gradle Increment annotation] Warning for test components.

### ISSUE THREAD

**Issue by karandeep-swiggy** [facebook/litho#655]
Hi, I am on the latest version of litho. Previously we use to get warnings for multiple litho annotations but now we are getting for only one.
`[WARN] Incremental annotation processing requested, but support is disabled because the following processors are not incremental: com.facebook.litho.specmodels.processor.testing.ComponentsTestingProcessor (NON_INCREMENTAL).`
Can someone from the team look into this. is this a false warning?

**Comment by muraziz** [comment]
@colriot does this sound related to recent robo v4 migration?

**Comment by karandeep-swiggy** [comment]
related: https://github.com/facebook/litho/pull/483

**Comment by colriot** [comment]
@muraziz nope. Migration will be reflected in the new release.
@karandeep-swiggy not sure what other warnings you were getting before, but getting "only one" seems better to me than "multiple" :) Briefly speaking, `ComponentsTestingProcessor` does not support incremental annotation processing and warning is correct. If you'll look at #483, you'll see that only production processors were changed to support incremental AP, but not testing processors. We would gladly accept a PR on that, if you want, but I'm not sure if this warning actually should affect you as it's only for tests using `@TestSpec`s

**Comment by pavlospt** [comment]
@colriot TestingProcessors were explicitly left out of IAP support. I can add support for the rest but I am not sure if this is needed. Also I am not sure why the OP is getting a warning about a *`TestingProcessor` in the first place 😛 !

@karandeep-swiggy a small repro project would be nice, in order to better check this issue!

**Comment by karandeep-swiggy** [comment]
@pavlospt I'll try to create a project and put the github link here. till then can we keep this issue open?

**Comment by colriot** [comment]
@pavlospt that was fast! 😆 
I saw this warning sometimes too. But I don't think this should affect the compilation time. Although, I'm not sure about the compilation of tests – wdyt?

**Comment by pavlospt** [comment]
@colriot I am generally against any kind of optimizations in test APs, unless it proves to be critical regarding compilation times and the cost of having false positives or weird errors when runnings tests, is bareable 😛

**Comment by colriot** [comment]
@pavlospt I totally agree with that! Heisenbugs due to the non-deterministic testing setup are the worst. But do we have an option to not show these warnings for specific APs?

**Comment by pavlospt** [comment]
> @pavlospt I totally agree with that! Heisenbugs due to the non-deterministic testing setup are the worst. But do we have an option to not show these warnings for specific APs?

I am not aware of any way to disable such warnings 😞
